src/kafka/kafka_clients.clj

```diff
@@ -0,0 +1,132 @@
+(ns kafka.kafka-clients
+  (:import (org.apache.kafka.clients.producer KafkaProducer ProducerRecord)
+           (org.apache.kafka.common.serialization StringSerializer StringDeserializer)
+           (org.apache.kafka.clients.consumer KafkaConsumer)
+           (java.time Duration)))
+
+(def configs
+  (atom {:producer-config {"value.serializer"  StringSerializer
+                           "key.serializer"    StringSerializer
+                           "client.id"         "m-w1kx3xw0nr"
+                           "bootstrap.servers" "localhost:9092"
+                           "acks"              "all"}
+         :consumer-config {"group.id"           "My-Group"
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer
+                           "client.id"          "m-w1kx3xw0nr"
+                           "bootstrap.servers"  "localhost:9092"}}))
+
+(defn build-producer
+  [configs]
+  (let [producer-config (@configs :producer-config)]
+    (new KafkaProducer producer-config)))
+
+(defn build-consumer
+  [configs]
```

Comment:
This is named as configs, but based on code below an atom is being passed?

Reply:
But we should avoid hungarian notation (i.e., configs-atom) right?
Should I pass the consumer config here (call this function with `(build-consumer (@config :consumer-config))`) ?